### PR TITLE
[RFC][UN-15295] Datepicker range select dates logic

### PR DESCRIPTION
**Pull request by niceoutput**
https://uniplaces.atlassian.net/browse/UN-15295

### CODE REVIEW

File: addon/components/uni-calendar.js [hunk 1]
```diff
@@ -1,8 +1,27 @@
...
   layout,
 
-  showAbove: false
+  showAbove: false,
+  startDate: moment(),
+  endDate: null,
+  showDatepicker: false,
+  minimumDays: 0,
+
+  dateRange: computed(function() {
```
Comment (DaniFrancisco) [inline addon/components/uni-calendar.js]:
```suggestion
  dateRange: computed('startDate', 'endDate', function() {
```

File: addon/components/uni-calendar.js [hunk 2]
```diff
@@ -1,8 +1,27 @@
...
+    return moment.range(this.get('startDate'), this.get('endDate'));
+  }),
+
+  actions: {
+    getDateAndOpenDatepicker(date, minimumDays) {
```
Comment (DaniFrancisco) [inline addon/components/uni-calendar.js]: `onStartDateSelected`

File: addon/components/uni-calendar.js [hunk 3]
```diff
@@ -1,8 +1,27 @@
...
+      let newDate = date.add('days', minimumDays);
```
Comment (DaniFrancisco) [inline addon/components/uni-calendar.js]:
```suggestion
      this.set('startDate', date);
      this.set('endDate',  date.add('days', minimumDays));
```

and remove the rest

File: addon/components/uni-datepicker-input.js [hunk 4]
```diff
@@ -34,14 +33,6 @@ export default Component.extend(ClickOutsideMixin, {
 
   onSelected() {},
 
-  onOutsideClick() {
```
Comment (DaniFrancisco) [inline addon/components/uni-datepicker-input.js]: is it working without this? 🤔

**Review by DaniFrancisco — CHANGES_REQUESTED** [review]
missing tests for the `minimumDays` feature

**Review by DaniFrancisco — APPROVED** [review]

**Review by DaniFrancisco — APPROVED** [review]

File: tests/integration/components/uni-calendar-test.js [hunk 5]
```diff
@@ -27,4 +28,18 @@ module('Integration | Component | uni calendar', function(hooks) {
 
     assert.dom('.uni-datepicker').doesNotExist();
   });
+
+  test('it renders the exact date chooosen plus the minimum days', async function(assert) {
```
Comment (sardyy) [inline tests/integration/components/uni-calendar-test.js]: fix typo with ```chooosen``` to ```chosen```

**Review by sardyy — APPROVED** [review]

**Review by henrmota — APPROVED** [review]

**Pull request merged**